renku/api/_git.py

```diff
@@ -197,23 +198,64 @@ def ensure_clean(self, ignore_std_streams=False):
             _clean_streams(self.repo, mapped_streams)
             raise errors.DirtyRepository(self.repo)
 
+    def ensure_untracked(self, path):
+        """Ensure that path is not part of git untracked files."""
+        untracked = self.repo.untracked_files
+
+        for file_path in untracked:
```

Comment:
sorry missed this the first time around - why do you need a for loop? Can't you just do

```
if file_path in untracked
```

Reply:
even better would be a set operation which would simplify the logic in `commit` below